pkg/reconciler/taskrun/taskrun.go

```diff
@@ -526,6 +526,21 @@ func (c *Reconciler) updateLabelsAndAnnotations(ctx context.Context, tr *v1beta1
 	}
 	tr.Annotations[podconvert.ReleaseAnnotation] = version
 
+	if c.SpireClient != nil {
+		if tr.IsSuccessful() {
+			if _, ok := tr.Annotations[spire.TaskRunStatusHashAnnotation]; !ok {
+				if c.SpireClient.AppendStatusAnnotation(tr) != nil {
+					return nil, fmt.Errorf("error appending taskRun %s status signed with private key: %w", tr.Name, err)
+				}
+			} else {
+				err := spire.CheckStatusAnnotationHash(tr)
```

Comment:
If the hash is not equivalent here, then we want to set the status annotation with `AppendStatusAnnotation`.. I.e. if there is a change in the task run status, the controller updates the hash and signature. Big question here is if the task run is immutable at this point, then we can totally skip the code, but if there's a valid use case, then we should update the hash/signature.

At least this was what my thinking was behind the comment. But if we think that once the controller says "Successful", then it is locked in then we don't need this check.

Reply:
ok I will make this log instead of error

Reply:
Commented this out for now. Need to ask Christie how to handle this case.